src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,435 @@
+     * after all arguments have been parsed. Tries to catch most cases where the user
+     * provides potentially confusing input.
+     */
+    @Override
+    public void validateArguments() throws CommandLineException {
+        // throw if a annotation is *enabled* more than once by the user
+        final Set<String> duplicateUserEnabledAnnotationNames = Utils.getDuplicatedItems(userArgs.getUserEnabledAnnotationNames());
+        if (!duplicateUserEnabledAnnotationNames.isEmpty()) {
+            throw new CommandLineException.BadArgumentValue(
+                    String.format("The annotation(s) are enabled more than once: %s",
+                            Utils.join(", ", duplicateUserEnabledAnnotationNames)));
+        }
+
+        // throw if a annotation is *disabled* more than once by the user
+        final Set<String> duplicateDisabledUserAnnotationNames = Utils.getDuplicatedItems(userArgs.getUserDisabledAnnotationNames());
+        if (!duplicateDisabledUserAnnotationNames.isEmpty()) {
+            throw new CommandLineException.BadArgumentValue(
+                    String.format("The annotation(s) are disabled more than once: %s",
+                            Utils.join(", ", duplicateDisabledUserAnnotationNames)));
+        }
+
+        // throw if a annotation is both enabled *and* disabled by the user
+        final Set<String> enabledAndDisabled = new HashSet<>(userArgs.getUserEnabledAnnotationNames());
+        enabledAndDisabled.retainAll(userArgs.getUserDisabledAnnotationNames());
+        if (!enabledAndDisabled.isEmpty()) {
+            final String badAnnotationList = Utils.join(", ", enabledAndDisabled);
+            throw new CommandLineException(
+                    String.format("The annotation(s): %s are both enabled and disabled", badAnnotationList));
+        }
+
+        // throw if a disabled annotation doesn't exist; warn if it wasn't enabled by the tool in the first place
+        userArgs.getUserDisabledAnnotationNames().forEach(s -> {
+            if (!allDiscoveredAnnotations.containsKey(s)) {
+                throw new CommandLineException.BadArgumentValue(String.format("Disabled annotation (%s) does not exist", s));
+            } else if (!toolDefaultAnnotations.containsKey(s)) {
+                logger.warn(String.format("Disabled annotation (%s) is not enabled by this tool", s));
+            }
+        });
+
+        // warn if an annotation is both default and enabled by the user
+        final Set<String> redundantAnnots = new HashSet<>(toolDefaultAnnotations.keySet());
+        redundantAnnots.retainAll(userArgs.getUserEnabledAnnotationNames());
+        redundantAnnots.forEach(
+                s -> {
+                    logger.warn(String.format("Redundant enabled annotation (%s) is enabled for this tool by default", s));
+                });
+
+        // warn if an annotation is both default and enabled by the user
```

Comment:
warn if an annotation -> warn if an annotation *group*

Reply:
Do we need to do anything special, or should we log, for the case where the user enables a group and redundantly enables an annotation in that group ?

Reply:
I suppose we could log, I didn't to it because it seems only natural in the case where you might have a group of annotations with arguments that come in from a group and you want to change those arguments, allowing you to type "-G SomeAnnotations -A AnAnnotation --argForAnAnnotation" which seams more natural than the alternative "-G SomeAnnotations --argForAnAnnotation". It doesn't seem like the command line should yell at you over the first case.